### PR TITLE
MMCA-4906 | Implement search error without search paramater on Cash Account dashboard

### PR DESCRIPTION
**Pull request by Neozxc**
Task:
- Add error validation for search input
- Update errors when system is down/timeouts

### CODE REVIEW

**Comment by AdamCoops** [comment]
You have conflicts with main, it'll be 4959 that's just merged earlier today.

File: app/controllers/CashAccountV2Controller.scala [hunk 1]
```diff
@@ -65,41 +65,49 @@ class CashAccountV2Controller @Inject()(authenticate: IdentifierAction,
...
+
+    val eventualMaybeCashAccount = apiConnector.getCashAccount(request.eori)
+    val (from, to) = cashAccountUtils.transactionDateRange()
+    val eventualStatements = getStatementsForEori(request.eori, from, to)
+    val transformedStatements = EitherT.liftF(eventualStatements)
+
+    val result = for {
+      cashAccount <- fromOptionF[Future, Result, CashAccount](eventualMaybeCashAccount, NotFound(eh.notFoundTemplate))
+      statements <- transformedStatements
+      page <-
```
Comment (AdamCoops) [inline app/controllers/CashAccountV2Controller.scala:94]: is it worth breaking this later in the line at like showAccountWithTransactionDetails( rather then on the <-?

I am not actually 100% sure what the default style we should be using here is

File: app/controllers/CashAccountV2Controller.scala
Comment (tyagikumarvineet) [inline app/controllers/CashAccountV2Controller.scala:1]: Could you please do the check (like below) at line after 116, whether form has error and then wrap the view in BadRequest as in that case its a BAD_REQUEST and should be presented in that way and should be different from OK?

        if (form.hasErrors) {
          BadRequest(accountsView(form, CashAccountV2ViewModel(req.eori, account, cashTransactions, statements, page)))
        } else {
          Ok(accountsView(form, CashAccountV2ViewModel(req.eori, account, cashTransactions, statements, page)))
        }

File: test/utils/RegexPatternsSpec.scala [hunk 2]
```diff
@@ -45,6 +45,28 @@ class RegexPatternsSpec extends SpecBase {
     }
   }
 
+  "UCR Regex" should {
```
Comment (tyagikumarvineet) [inline test/utils/RegexPatternsSpec.scala:48]: Better to maintain the order of the tests same as the vals are in the object. So better to move these tests after "Payment Regex"

File: app/controllers/CashAccountV2Controller.scala (hunk 1 above)
Comment (Neozxc) [reply app/controllers/CashAccountV2Controller.scala]: Reformatted per your comment

**Review by AdamCoops — APPROVED** [review]

**Review by tyagikumarvineet — APPROVED** [review]

**Pull request merged**